src/libostree/ostree-repo.c

```diff
@@ -3771,7 +3771,20 @@ load_metadata_internal (OstreeRepo       *self,
           if (!glnx_fstatat_allow_noent (self->repo_dir_fd, commitpartial_path, NULL, 0, error))
             return FALSE;
           if (errno == 0)
-            *out_state |= OSTREE_REPO_COMMIT_STATE_PARTIAL;
+            {
+              glnx_autofd int fd = -1;
+              ot_openat_ignore_enoent (self->repo_dir_fd, commitpartial_path, &fd, error);
```

Comment:
Still need to do:
```
if (!ot_openat_ignore_enoent (...))
  return FALSE;
```

The idea is that we still want to surface *other* I/O errors like `EIO`, `EPERM` etc.

Reply:
Actually we could just stop doing the `stat()` call above too; no reason to do `stat()` as an existence check + `open()` when one can just do `open()` and check for `ENOENT`.